### PR TITLE
[17.0][ADD] repair_type_product_destination: New module repair_type_product_destination

### PR DESCRIPTION
**Pull request by ppyczko**
This is a technical module, and it does not provide any new functionality on its own. Due to the way transfers of repaired products work in Odoo, the following fields will not have any effect without additional code modifications.

This module introduces a **Product Destination Location field** in the repair order. The default value for this field is computed from the **Default Product Destination Location field of the associated Operation Type**.

This module essentially serves as a template, providing the groundwork for further customization. To make use of the functionality it introduces, extend this module to implement additional features in the repair transfer workflow.

cc https://github.com/APSL 160548

@miquelalzanillas @lbarry-apsl @mpascuall @peluko00 @javierobcn @BernatObrador please review

### CODE REVIEW

File: repair_type_product_destination/models/repair.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class RepairOrder(models.Model):
+    _inherit = "repair.order"
+
+    product_location_dest_id = fields.Many2one(
+        "stock.location",
+        "Product Destination Location",
+        compute="_compute_product_location_dest_id",
+        store=True,
+        readonly=False,
```
Comment (peluko00) [inline repair_type_product_destination/models/repair.py]: Remove that please. It's no take effect

**Review by peluko00 — CHANGES_REQUESTED** [review]

File: repair_type_product_destination/models/repair.py (hunk 1 above)
Comment (ppyczko) [reply repair_type_product_destination/models/repair.py]: Done! Thanks @peluko00

**Review by peluko00 — APPROVED** [review]
LGTM, code review and tested in runboat

**Comment by peluko00** [comment]
@LoisRForgeFlow can you review that please? Thanks!

**Review by mpascuall — APPROVED** [review]
LGTM, tested in runboat

**Review by BernatObrador — APPROVED** [review]
LGTM, tested in runboat and code review

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

**Pull request merged**